src/user/views/custom_verify_email_view.py

```diff
@@ -0,0 +1,60 @@
+from allauth.account.models import EmailConfirmation, EmailConfirmationHMAC
+from dj_rest_auth.registration.views import VerifyEmailView
+from django.contrib.auth import get_user_model
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+
+User = get_user_model()
+
+
+class CustomVerifyEmailView(VerifyEmailView):
+    def post(self, request, *args, **kwargs):
+
+        key = request.data.get("key")
+
+        if not key:
+            return Response({"detail": "Invalid key"}, status=400)
+
+        # Store user info BEFORE calling parent method
+        # because the parent method will override it
+        user_id = None
+
+        confirmation = EmailConfirmationHMAC.from_key(key)
+
+        if not confirmation:
+            confirmation = EmailConfirmation.objects.filter(key=key.lower()).first()
+
+        if confirmation:
+            user = confirmation.email_address.user
+            user_id = user.id
+
+        # Call parent method to handle verification
+        parent_response = super().post(request, *args, **kwargs)
+
+        # If parent verification was successful AND we have user info
+        if parent_response.status_code == 200 and user_id:
```

Comment:
Nit: Instead of nesting the main logic by the conditional `if` and the `try`/`except`, it might be easier to return here in case of errors ("align happy path to the left").